docs/source/llms/openai/notebooks/openai-embeddings-generation.ipynb

```diff
@@ -0,0 +1,513 @@
+    "\n",
+    "assert \"OPENAI_API_KEY\" in os.environ, \" OPENAI_API_KEY environment variable must be set\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Integrating OpenAI Model with MLflow for Document Similarity\n",
+    "\n",
+    "In this tutorial segment, we demonstrate the process of setting up and utilizing an OpenAI embedding model within MLflow for document similarity tasks.\n",
+    "\n",
+    "#### Key Steps:\n",
+    "\n",
+    "1. **Set MLflow Experiment**: We begin by setting the experiment context in MLflow, specifically for document similarity, using `mlflow.set_experiment(\"Documentation Similarity\")`.\n",
+    "\n",
+    "2. **Logging the Model in MLflow**: We initiate an MLflow run and log the OpenAI model. The model in focus is \"text-embedding-ada-002\", chosen for its robust embedding capabilities. During this step, we detail the model, the embedding task, input/output schemas, and parameters like batch size.\n",
+    "\n",
+    "3. **Loading the Logged Model for Use**: After logging the model, we proceed to load it using MLflow's `pyfunc` module. This is a critical step for applying the model to perform document similarity tasks within the MLflow ecosystem.\n",
+    "\n",
+    "These steps are essential for integrating OpenAI's embedding model into MLflow, facilitating advanced NLP operations like document similarity analysis.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2023/12/14 13:32:42 INFO mlflow.tracking.fluent: Experiment with name 'Documenatation Similarity' does not exist. Creating a new experiment.\n"
+     ]
+    }
+   ],
+   "source": [
+    "mlflow.set_experiment(\"Documenatation Similarity\")\n",
+    "\n",
+    "with mlflow.start_run():\n",
+    "    model_info = mlflow.openai.log_model(\n",
+    "        model=\"text-embedding-ada-002\",\n",
+    "        task=openai.Embedding,\n",
+    "        artifact_path=\"model\",\n",
+    "        signature=ModelSignature(\n",
+    "            inputs=Schema([ColSpec(type=\"string\", name=None)]),\n",
+    "            outputs=Schema([TensorSpec(type=np.dtype(\"float64\"), shape=(-1,))]),\n",
+    "            params=ParamSchema([ParamSpec(name=\"batch_size\", dtype=\"long\", default=1024)]),\n",
+    "        ),\n",
+    "    )\n",
+    "\n",
+    "# Load the model in pyfunc format\n",
+    "model = mlflow.pyfunc.load_model(model_info.model_uri)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Webpage Text Extraction for Embedding Analysis\n",
+    "\n",
+    "This section of the tutorial introduces functions designed to extract and prepare text from webpages, a crucial step before applying embedding models for analysis.\n",
+    "\n",
+    "#### Overview of Functions:\n",
```

Comment:
I feel this section (and similar ones in other cells) is redundant to docstring, can we remove either one?

Reply:
Removed the actual functions in favor of using sklearn's implementations :)

Reply:
Adding to this one - can we also simplify the function description and move Detailed Workflow after the code? My first impression is that reading this text is a bit hard without seeing the code first. Non-blocking tho.